chain/client/tests/process_blocks.rs

```diff
@@ -914,14 +890,16 @@ fn test_gc_long_epoch() {
     let mut blocks = vec![];
 
     for i in 1..=num_blocks {
-        let block_producer = env.clients[0]
-            .runtime_adapter
-            .get_block_producer(&EpochId(CryptoHash::default()), i)
-            .unwrap();
-        if block_producer == "test0".to_string() {
-            let block = env.clients[0].produce_block(i).unwrap().unwrap();
-            env.process_block(0, block.clone(), Provenance::PRODUCED);
-            blocks.push(block);
+        if i < epoch_length || i == num_blocks {
```

Comment:
why?

Reply:
Because we need to have a very long epoch, and the epoch switches when the previous block has the height over the epoch boundary. It worked before because before the epoch would only switch when there's a final block.